### ISSUE TITLE
[MSSQL] Connection issues with Azure SQL database

### ISSUE THREAD

**Issue by bor-attila** [eclipse-vertx/vertx-sql-client#978]
### Version

4.1.1-SNAPSHOT

### Steps to reproduce

1. I simply created a test maven project with start.vertx.io
2. Checked Reactive MSSQL client  as  dependency
3. Used the sample code from the [documentation](https://vertx.io/docs/vertx-mssql-client/java/#_getting_started)
4. Changed credentials (username/pass etc)

Instead of result I got an error message.

**Failure: Fail to read any response from the server, the underlying connection might get lost unexpectedly.**

### Extra

* java --version
openjdk 16 2021-03-16
OpenJDK Runtime Environment (build 16+36-2231)
OpenJDK 64-Bit Server VM (build 16+36-2231, mixed mode, sharing)
* OS Windows 10
* [JDBC](https://vertx.io/docs/vertx-jdbc-client/java/) works great

**Comment by tsegismont** [comment]
Thanks for reporting, we'll investigate asap

Le jeu. 10 juin 2021 à 03:31, Attila Bőr ***@***.***> a
écrit :

**Comment by tsegismont** [comment]
@bor-attila I suspect this is due to #605 

But I won't close as duplicate until we can make sure resolving #605 allows to connect to an Azure SQL db

**Comment by bor-attila** [comment]
Yes, I think the problem is the TLS, but I have no "proof for that". 

I tried to increate timeout settings, but no luck.

A jdbc string for azure sql server looks like this:

**jdbc:sqlserver://{host}:1433;database={database};user={user};password={your_password_here};encrypt=true;trustServerCertificate=false;hostNameInCertificate=*.database.windows.net;loginTimeout=30;**

There are some encryption settings, imo with a very good reason : )

**Comment by tsegismont** [comment]
@bor-attila with #1017 we will get encryption support for the MS SQL Client.

I did some testing with Azure SQL Server. The client is able to connect from the outside (proxy mode) but node from the inside (redirect mode). I did some debugging and I believe we need to support the TDS `ROUTING` token (which allows a server to redirect a client to another server).

See https://github.com/eclipse-vertx/vertx-sql-client/pull/1017#issuecomment-896565891
